### PR TITLE
Enable precompiling CoreRoot overlay with crossgen for Linux arm/arm64

### PR DESCRIPTION
**Pull request by sergiy-k**
Fixes https://github.com/dotnet/coreclr/issues/23815

### CODE REVIEW

**Comment by echesakovMSFT** [comment]
@sergiy-k @RussKeldorph I pushed 3 commits enabling R2R testing against this pull request.

Please undo them before merging this PR - I am planning on enabling R2R testing in my another PR https://github.com/dotnet/coreclr/pull/23848 along with workaround for xunit failing R2R on Linux/macOS right now.

File: build-test.sh [hunk 1]
```diff
@@ -149,8 +149,8 @@ generate_layout()
 
     build_MSBuild_projects "Tests_Overlay_Managed" "${__ProjectDir}/tests/runtest.proj" "Creating test overlay" "/t:CreateTestOverlay"
 
-    chmod +x $__BinDir/corerun
-    chmod +x $__BinDir/crossgen
+    chmod +x $__CrossgenExe
```
Comment (echesakovMSFT) [inline build-test.sh]: I think this line shouldn't be changed

Comment (RussKeldorph) [reply build-test.sh]: oh yeah good catch

**Pull request merged**